File: credit.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>

string flag = "INVALID";

int contardigitos, digitos, restosum = 0;
long cardnumber;

int main(void)
{
    cardnumber = get_long("Enter your cardnumber:\n");   
    //printf("\n\ndigitos: %i\nContarDigitos: %i\nCardnumber: %li\n", digitos,contardigitos,cardnumber);

    restosum = cardnumber % 10;
    cardnumber = cardnumber / 10;

    do
    {   
        if (((cardnumber % 10) * 2) >= 10)
        {
            contardigitos = contardigitos + (((cardnumber % 10) * 2) - 9);
            cardnumber = cardnumber / 10;
            digitos++;
        }
        else
        {
            contardigitos = contardigitos + (cardnumber % 10) * 2;
            cardnumber = cardnumber / 10;
            digitos++; 
        }

        restosum = cardnumber % 10 + restosum;
        cardnumber = cardnumber / 10;
        digitos++;

        if ((cardnumber == 37 || cardnumber == 34) || ((cardnumber <= 379 && cardnumber >= 370) || (cardnumber >= 340 
                && cardnumber <= 349))) 
        {
            flag = "a"; //AMEX
            digitos++;
        }
 
        if ((cardnumber <= 55 && cardnumber >= 51) || (cardnumber <= 559 && cardnumber >= 510))
        {
            flag = "m"; //MASTER
        }

        if ((cardnumber <= 49 && cardnumber >= 40) || (cardnumber <= 499 && cardnumber >= 400))
        {
            flag = "v";
        }    
    }
    
    while (cardnumber >= 1);

    cardnumber = restosum + contardigitos;

    if ((cardnumber % 10) != 0)
    {
        flag = "INVALID";
    }

    if ((strcmp(flag, "a") == 0) && digitos == 15) 
    {
        flag = "AMEX";
    }
    else if ((strcmp(flag, "m") == 0) && digitos == 16)
    {
        flag = "MASTERCARD"; 
    }
    else if ((strcmp(flag, "v") == 0) && (digitos == 13 || digitos == 16))
    {
        flag = "VISA";
    }
    else 
    {
        flag = "INVALID";
    }

    // printf("Digitos: %i\nContarDigitos: %i\nRestosum: %i\nCardnumber: %li\nFlag: %s\n\n\n", digitos,contardigitos, restosum, cardnumber,flag);
    printf("%s\n", flag);
}

File: credit.py
from cs50 import get_int


flag = "INVALID"

contardigitos = 0
digitos = 0
restosum = 0


cardnumber = get_int("Enter your cardnumber:\n")

# printf("\n\ndigitos: %i\nContarDigitos: %i\nCardnumber: %li\n", digitos,contardigitos,cardnumber)

restosum = cardnumber % 10
cardnumber = cardnumber // 10


while cardnumber >= 1:

    if ((cardnumber % 10) * 2) >= 10:

        contardigitos = contardigitos + (((cardnumber % 10) * 2) - 9)
        cardnumber = cardnumber // 10
        digitos += 1

    else:

        contardigitos = contardigitos + (cardnumber % 10) * 2
        cardnumber = cardnumber // 10
        digitos += 1

    restosum = cardnumber % 10 + restosum
    cardnumber = cardnumber // 10
    digitos += 1

    if ((cardnumber == 37 or cardnumber == 34) or ((cardnumber <= 379 and cardnumber >= 370) or (cardnumber >= 340 and cardnumber <= 349))):

        flag = "a"
        digitos += 1

    if ((cardnumber <= 55 and cardnumber >= 51) or (cardnumber <= 559 and cardnumber >= 510)):

        flag = "m"

    if ((cardnumber <= 49 and cardnumber >= 40) or (cardnumber <= 499 and cardnumber >= 400)):

        flag = "v"


cardnumber = restosum + contardigitos

if ((cardnumber % 10) != 0):

    flag = "INVALID"


if (flag == "a") and digitos == 15:

    flag = "AMEX"

elif (flag == "m") and digitos == 16:

    flag = "MASTERCARD"

elif (flag == "v") and (digitos == 13 or digitos == 16):

    flag = "VISA"

else:

    flag = "INVALID"


print("" + flag)


#print(f"Digitos: {digitos} \nContarDigitos: {contardigitos}\nRestosum:{restosum}\nCardnumber: {cardnumber}\nFlag: {flag}\n\n\n")

